api/client.go

```diff
@@ -327,6 +331,10 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 		}
 	case *charm.CharmArchive:
 		var err error
+		// For local charm check if it is revision control directory
```

Comment:
Comment can be expanded a bit  - we're generating a version file not just checking it is a vcs dir

Reply:
Also, a charm archive is a zip file. There's no need to generate a version file for that case